### DISCUSSION TITLE
No equals and hashcode?

### DISCUSSION THREAD

**Discussion by igaliash-evooq** [cloudbees/zendesk-java-client#469] [Q&A] [answered]
Is there any reason why domain model classes do not have equals and hashcode implementations? It makes unit testing pretty painful.

**Comment by aheritier** [comment, marked as answer]
No good idea on my side. I think we could add them but in general they will just check the ID ?

**Reply by igaliash-evooq** [reply]
There are places where you don't even have an ID yet, e.g. when you create a ticket. How would you check that request you are sending is the same as it supposed to be without verifying all the fields? 
If for any reason I'd like to check IDs only it should be pretty easy to do.

**Reply by PierreBtz** [reply]
It seems that you are trying to test that the `zendesk-java-client` behaves properly, which shouldn't be the concern of a project consuming it. Do you have an example of a test that causes you trouble, if it's indeed testing the `zendesk-java-client` we could maybe port it directly into the test suite of the client, it would relieve you of the pain.
